content/docs/fundamentals/converting/resolving-errors@id.md

```diff
@@ -0,0 +1,350 @@
+Setiap dokumen AMP harus memiliki link yang merujuk pada versi "kanonis" dokumen tersebut.  Kita akan mempelajari lebih lanjut apa itu halaman kanonis dan berbagai pendekatan terhadap penautan kanonis dalam langkah [Membuat halaman Anda dapat ditemukan](/id/docs/fundamentals/converting/discoverable.html) di tutorial ini.
+
+Untuk tutorial ini, kita akan menggunakan artikel HTML asli yang dikonversi menjadi halaman kanonis.
+
+Lanjutkan dan **tambahkan** kode berikut di bawah tag `<meta charset="utf-8" />`:
+
+```html
+<link rel="canonical" href="/article.html">
+```
+
+{% call callout('Catatan', type='note') %}
+Anda dapat membuat halaman AMP kanonis mandiri. Link kanonis masih diperlukan, tetapi harus mengarah ke artikel AMP itu sendiri:
+
+```html
+<link rel="canonical" href="article.amp.html">
+```
+{% endcall %}
+
+Sekarang, **muat ulang** halaman. Meskipun masih ada banyak error yang perlu diperbaiki, error pada link kanonis sudah tidak ada.
+
+## Tentukan atribut AMP
+
+AMP memerlukan atribut di elemen `<html>` utama halaman untuk menyatakan halaman sebagai dokumen AMP.
+
+<pre class="error-text">
+Atribut wajib '⚡' tidak ada di tag 'html ⚡ for top-level html'
+Tag wajib 'html ⚡ for top-level html' tidak ada atau salah.
+</pre>
+
+Error di atas dapat diatasi hanya dengan menambahkan atribut `⚡` ke tag `<html>` seperti berikut:
+
+```html
+<html ⚡ lang="en">
+```
+
+Sekarang, lanjutkan, muat ulang halaman, dan pastikan kedua error sudah tidak ada.
+
+{% call callout('Catatan', type='note') %}
+Meskipun direkomendasikan untuk menentukan `⚡`, kita juga dapat menggunakan atribut `amp` untuk menggantikan atribut `⚡`, seperti yang berikut:
+
+```html
+<html amp lang="en">
+```
+{% endcall %}
+
+## Tentukan viewport
+
+Selanjutnya, mari kita selesaikan error berikut:
+
+<pre class="error-text">
+Tag wajib 'meta name=viewport' tidak ada atau salah.
+</pre>
+
+AMP memerlukan definisi `width` dan `minimum-scale` untuk viewport. Nilai ini masing-masing harus ditetapkan sebagai `device-width` dan `1`. Viewport adalah tag umum yang disertakan dalam `<head>` pada halaman HTML.
+
+Untuk menyelesaikan error viewport, tambahkan cuplikan HTML berikut ke tag `<head>`:
+
+```html
+<meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+```
+
+Nilai untuk `width` dan `minimum-scale` wajib ditetapkan di AMP. Menetapkan `initial-scale` tidak wajib dilakukan, tapi biasanya disertakan dalam pengembangan web seluler dan hal itu direkomendasikan. Anda dapat membaca selengkapnya tentang viewport dan desain yang responsif di [Mengonfigurasi Area Pandang](https://developers.google.com/speed/docs/insights/ConfigureViewport).
+
+Seperti sebelumnya, **muat ulang** halaman dan periksa apakah error sudah tidak muncul.
+
+## Ganti stylesheet eksternal
+
+Error berikut berkaitan dengan penggunaan kita atas stylesheet:
+
+<pre class="error-text">
+Atribut 'href' di tag 'link rel=stylesheet for fonts' ditetapkan ke nilai yang tidak valid 'base.css'.
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).